### PR TITLE
PP-10512 Delete Stored Payment Details (Payment Provider method)

### PR DESCRIPTION
**Pull request by james-peacock-ho**
- add default deleteStoredPaymentDetails method to PP interface
- call deleteStoredPaymentDetails method from task handler
- integration test
- Worldpay implementation will follow in separate PR

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return GatewayOperation.DELETE_STORED_PAYMENT_DETAILS;
+    }
+
+    @Override
+    public Map<String, Object> getGatewayCredentials() {
+        return agreement.getGatewayAccount().getCredentials(agreement.getGatewayAccount().getGatewayName());
+    }
+
+    @Override
+    public AuthorisationMode getAuthorisationMode() {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java:44]: I don't think we need this.

**Review by stephencdaly — APPROVED** [review]
LGTM! Just a getter that I don't think will be used which can be removed in next PR.

File: src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java (hunk 1 above)
Comment (james-peacock-ho) [reply src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java:44]: I agree we shouldn't need it, but I think we have to include it if we are implementing GatewayRequest interface.

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java:44]: Oh - do we need to implement that interface?

Comment (james-peacock-ho) [reply src/main/java/uk/gov/pay/connector/gateway/model/request/DeleteStoredPaymentDetailsGatewayRequest.java:44]: ha ha, good question!

**Pull request merged**